### PR TITLE
Fix #164: Implement Buccaneer Valiant special ability

### LINKED ISSUE #164: The Broadsider's schooner won't work
### 

- [X] I have checked that this issue has not already been reported.

- [X] I have validated that this issue is not a syntax error of either MySQL or SQLite.

- [X] I have pulled the latest version of the main branch of DarkflameServer and have confirmed that the issue exists there.

### Issue Description

The schooner for the Buccaneer's Broadsider spawns in but does not do any damage to the enemy. Also on the Ubuntu command prompt there is this issue [CppScripts]: Attempted to load CppScript for 'scripts\EquipmentScripts\BuccaneerValiantShip.lua', but returned InvalidScript. 

### Reproduction steps

Get the Broadsider and use the third ability

### Expected Behavior

I expected the schooner to ram the enemy but it just sits there doing nothing 

### Environment

I was running it on Ubuntu.

### PR DESCRIPTION
**Pull request by MickVermeulen**
Adds the ability for the buccaneer valiant to spawn a ship that rams
enemies and smashes them. Next to a script that triggers the ship skill
a few other changes had to be made:
- Force movement behavior server side calculation and sync
- The ship has no physics volume so the FindValidTargets for behaviors
had to be altered to allow ControllablePhysics entities to find entities
within their area. The "target_self" AOE flag has been used to replicate
the old behavior.

Attached is a video of the new behavior. As there were no variables in the BehaviorParameter table wrt the speed of the ship I'm curious to know if the current speed feels right.

Also keen on Wincent and Xipho reviewing the behavior changes I made. I wasn't sure what to test those on. If you have any suggestions let me know.

https://user-images.githubusercontent.com/22911950/145674133-a66d9a93-c079-4307-a3cc-94ab8c262a62.mp4

### CODE REVIEW

**Comment by MickVermeulen** [comment]
Not sure why the title didn't trigger this but fixes  #164.

**Comment by MickVermeulen** [comment]
So @Wincent01 , or @Xiphoseer , any opinions?

**Comment by Wincent01** [comment]
Looks good to me

**Review by Xiphoseer — APPROVED** [review]
Yeah, looks good to me as well. The AoE target logic may be missing some more flags as per <https://docs.lu-dev.net/en/latest/game-mechanics/behaviors/aoe.html> (generated from the DB), but I think this is a net improvement.

File: dScripts/BuccaneerValiantShip.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#include "BuccaneerValiantShip.h"
+#include "SkillComponent.h"
+#include "dLogger.h"
+
+void BuccaneerValiantShip::OnStartup(Entity* self) {
+    const auto skill = 982;
+    const auto behavior = 20577;
+    const auto skillCastTimer = 1.0F;
+
+    self->AddCallbackTimer(skillCastTimer, [self]() {
```
Comment (Wincent01) [inline dScripts/BuccaneerValiantShip.cpp:10]: Have to capture skill and behavior here if you want to use them within the lambda. The current implementation isn’t standard.

**Pull request merged**